compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -446,9 +446,13 @@ abstract class PrepJSInterop[G <: Global with Singleton](val global: G)
           }
         }
 
-        val exports = exporters.get(clsSym).toIterable.flatten
-        // Add exports to the template
-        treeCopy.Template(tree, parents, self, body ++ exports)
+        // Add exports to the template, if there are any
+        exporters.get(clsSym) match {
+          case None =>
+            tree // nothing to change
+          case Some(exports) =>
+            treeCopy.Template(tree, parents, self, body ::: exports.toList)
+        }
```

Comment:
Wouldn't you want to use a fold for this?

Reply:
Indeed. Updated.